pkg/virt-controller/watch/vmi.go

```diff
@@ -254,8 +257,22 @@ func (c *VMIController) execute(key string) error {
 	if !controller.ObservedLatestApiVersionAnnotation(vmi) {
 		vmi := vmi.DeepCopy()
 		controller.SetLatestApiVersionAnnotation(vmi)
+		key := controller.VirtualMachineInstanceKey(vmi)
+		c.vmiExpectations.SetExpectations(key, 1, 0)
 		_, err = c.clientset.VirtualMachineInstance(vmi.ObjectMeta.Namespace).Update(vmi)
-		return err
+		if err != nil {
+			c.vmiExpectations.LowerExpectations(key, 1, 0)
+			return err
+		}
+		return nil
+	}
+
+	// If needsSync is true (expectations fulfilled) we can make save assumptions if virt-handler or virt-controller owns the pod
+	needsSync := c.podExpectations.SatisfiedExpectations(key) && c.vmiExpectations.SatisfiedExpectations(key)
+	needsStatusUpdate := c.vmiExpectations.SatisfiedExpectations(key)
+
+	if !needsSync && !needsStatusUpdate {
```

Comment:
I am not entirely sure about this condition here. I think that we should simply just continue if all expectations are fulfilled.

Basically a

```
if !needsSync {
 return nil
}
```

I don't think it makes much sense to update the VMI status while we wait for the pod expectation to be fulfilled. That should make `needsStatusUpdate` checks unnecessary later on. What do you think?

Reply:
yup, fixed